### PR TITLE
Correct fix for #3672

### LINKED ISSUE #3659: Toolbar middleware breaks custom logout template
If you customize the standard `registration/logged_out.html` template, this cannot inherit from a base template that has `{% cms_toolbar %}`. If you do, you'll get:

    TypeError: int() argument must be a string or a number, not 'AnonymousUser'

If you refresh the page, the error goes away. This is because the middleware sets attributes on the request object during process_request at which point the user is still logged in because the logout view hasn't been called yet. The tag picks up from there assuming all is good but of course is not.

### PR DESCRIPTION
**Pull request by yakky**
Fix #3672 
Fix #3659

### CODE REVIEW

**Comment by yakky** [comment]
This will require a different implementation for 3.1

**Comment by yakky** [comment]
@mkoistinen would you check this? I'd like to avoid another regression in the toolbar for the same bug :D

**Comment by jsma** [comment]
I just tried to reproduce #3659 on the latest in the develop branch (haven't tried this PR) but I can't reproduce the bug anymore, logging out works. I haven't followed recent developments in develop the past few weeks but perhaps something there fixed this issue already? Or were your tests here failing without your patch to toolbar.py? Either way, thanks! 

I started working on a test case for this but turned into an exercise in yak shaving (mostly due to my unfamiliarity with how the django-cms test suite is organized) and wasn't able to complete. Looks like I can delete that git stash now and I can learn from your example here instead for next time, thanks again!

**Comment by yakky** [comment]
Actually no changes directly related to this in develop.
I'll rebase the test to check.
The toolbar framework will go through a bit of refactoring in develop, so thi will be fixed in quite a different way

File: cms/test_utils/project/templates/registration/logged_out.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "base.html" %}
+{% load i18n %}
+
+
+{% block content %}
+
+<p>{% trans "Thanks for spending some quality time with the Web site today." %}</p>
```
Comment (mkoistinen) [inline cms/test_utils/project/templates/registration/logged_out.html]: Some may see this as "passive aggressive". "Web site" should probably just be "website".

Perhaps we should just say something to confirm the action: "You have successfully logged-out." perhaps?

Comment (yakky) [inline cms/test_utils/project/templates/registration/logged_out.html]: This is just the standard django logout message 
This template it's just for testing purposes, it's never shown to the user

**Comment by jsma** [comment]
Haven't investigated further yet but did notice that when I logout, the toolbar appears on the logged out template with the login form displayed in the toolbar. Navigate to another page, the toolbar is cleared. So maybe *something* changed the logic so it smartly shows the login form instead of assuming a logged in user which resulted in the exception I experienced previously. Again, this is only on the develop branch.

**Comment by jsma** [comment]
As mentioned above the bug was apparently fixed through an unrelated change. It's been too long since I've looked at this code so have no comment on the small change to the "is_staff" check but the tests will be good to have to prevent regressions.

**Pull request merged**